homeassistant/components/youless/__init__.py

```diff
@@ -0,0 +1,53 @@
+"""The youless integration."""
+import asyncio
+from urllib.error import URLError
+
+import voluptuous as vol
+from youless_api import YoulessAPI
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import PlatformNotReady
+
+from .const import DOMAIN
+
+CONFIG_SCHEMA = vol.Schema({DOMAIN: vol.Schema({})}, extra=vol.ALLOW_EXTRA)
+PLATFORMS = ["sensor"]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the youless component."""
+    hass.data[DOMAIN] = {}
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up youless from a config entry."""
+    try:
+        hass.data[DOMAIN][entry.entry_id] = YoulessAPI(entry.data[CONF_HOST])
```

Comment:
You're doing IO in async context which is not allowed, it is also not recommended for a lib to do IO as part of the creation of the object

Reply:
I'll change this around a bit so that the IO does not happen in the integration initialization. In the config_flow a similar IO check is done to verify that the IP provided actually exists. Is this type OK.

Reply:
If you can wrap it in such a way that you can call it with async_add_executor_job it should be ok

Reply:
I've changed it so there is no longer a direct IO call from the async methods.